homeassistant/components/renson/__init__.py

```diff
@@ -56,3 +57,59 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         hass.data[DOMAIN].pop(entry.entry_id)
 
     return unload_ok
+
+
+def setup_hass_services(hass: HomeAssistant, renson_api: RensonVentilation) -> None:
+    """Set up the Renson platforms."""
+
+    async def set_timer_level(call: ServiceCall) -> None:
+        """Set timer level."""
+        level_string = call.data.get("timer_level", "Level1")
+        time = call.data.get("time", 0)
+        level = Level[level_string.upper()]
+
+        await hass.async_add_executor_job(renson_api.set_timer_level, level, time)
+
+    async def set_breeze(call: ServiceCall) -> None:
+        """Configure breeze feature."""
+        level = call.data.get("breeze_level", "")
+        temperature = call.data.get("temperature", 0)
+        activated = call.data.get("activate", False)
+
+        await hass.async_add_executor_job(
+            renson_api.set_breeze, level, temperature, activated
+        )
+
+    async def set_day_night_time(call: ServiceCall) -> None:
+        """Configure day night times."""
+        day = call.data.get("day", "7:00")
+        night = call.data.get("night", "22:00")
+
+        await hass.async_add_executor_job(renson_api.set_time, day, night)
+
+    async def set_pollution_settings(call: ServiceCall) -> None:
+        """Configure pollutions settings."""
+        day = call.data.get("day_pollution_level", "")
+        night = call.data.get("night_pollution_level", "")
+        humidity_control = call.data.get("humidity_control", "")
+        airquality_control = call.data.get("airquality_control", "")
+        co2_control = call.data.get("co2_control", "")
+        co2_threshold = call.data.get("co2_threshold", 0)
+        co2_hysteresis = call.data.get("co2_hysteresis", 0)
+
+        await renson_api.set_pollution(
+            day,
+            night,
+            humidity_control,
+            airquality_control,
+            co2_control,
+            co2_threshold,
+            co2_hysteresis,
+        )
+
+    hass.services.async_register(DOMAIN, "set_breeze", set_breeze)
+    hass.services.async_register(DOMAIN, "set_day_night_time", set_day_night_time)
+    hass.services.async_register(
+        DOMAIN, "set_pollution_settings", set_pollution_settings
+    )
+    hass.services.async_register(DOMAIN, "set_timer_level", set_timer_level)
```

Comment:
Please add the schema for all services

Reply:
I cannot find a service that is not in the service.yaml file?

Reply:
`services.yaml` is providing context to frontend. @edenhaus refers to a schema here in the code which is doing the validation of that input from the frontend when the user is calling the services.
Example for this one could be (not working):
```python
hass.services.async_register(
    DOMAIN, 
    "set_timer_level",
    set_timer_level
    {
        vol.Required(CONF_TIMER_LEVEL): vol.In(.......),
        vol.Required(CONF_TIME): cv.time,
)
```